assets/tests/TaskSchemaValidationService.js

```diff
@@ -0,0 +1,262 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for validating data put into a task schema
+ * within a question, before that schema is committed.
+ */
+
+tie.factory('TaskSchemaValidationService', [
+  function() {
+    return {
+      verifyInstructionsAreNotEmpty: function(task) {
+        var instructions = task.getInstructions();
+        return instructions.length > 0;
+      },
+      verifyInstructionsIsArrayOfStrings: function(task) {
+        var instructions = task.getInstructions();
+        if (!Array.isArray(instructions)) {
+          return false;
+        }
+        for (var i = 0; i < instructions.length; i++) {
+          var instruction = instructions[i];
+          if (!(typeof instruction === 'string' ||
+            instruction instanceof String)) {
```

Comment:
I'm a bit worried about using every, but after using it a few more times I feel like I get it. I'll update the tests with the changes.